airflow/models/baseoperator.py

```diff
@@ -215,6 +217,7 @@ def partial(
     start_date: datetime | ArgNotSet = NOTSET,
     end_date: datetime | ArgNotSet = NOTSET,
     owner: str | ArgNotSet = NOTSET,
+    deferrable: bool | ArgNotSet = NOTSET,
```

Comment:
I'm still not sure how I feel about putting this on BaseOperator. It really feels like, for example, `BashOperator(..., deferrable=True)` should raise still.

Shooting from the hip here, but maybe we add a `BaseDeferrableOperator` or something instead, and when we have 2.7+ for providers, providers can switch over to it?

Reply:
Or a `DeferrableMixin` (whichever is easier)

Reply:
Thanks @uranusjr for your suggestion! Just had a short discussion with @jedcunningham . I'll go with `DeferrableMixin`

Reply:
I just had another discussion with @uranusjr earlier today. We think `BaseDeferrableOperator` might be a better solution instead. Thus, I'll try to chang it accordingly early tomorrow

Reply:
Hi @jedcunningham @uranusjr , I just finished the `BaseDeferrableOperator` implementation. Could you please check when you're available? Thanks! 🙂